### ISSUE TITLE
md-select throws TypeError: undefined function when 'mode' property exists in scope

### ISSUE THREAD

**Issue by CameronHall** [angular/material#9230]
**Actual Behavior**:
 -  `What is the issue? *` md-select throws TypeError: undefined function when 'mode' property exists in scope
```
TypeError: undefined is not a function at Array.map (native) at SelectMenuController.self.selectedLabels
```
 -  `What is the expected behavior?` An error will not be thrown

**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:` http://codepen.io/anon/pen/NAOWNG
 -  `Details:`  If you click the md-select box whilst looking at the console you can see the error. If you change the '$scope.mode' to something else e.g. '$scope.isCreate' the error no longer appears.

**Angular Versions**: *
 -  `Angular Version:` 1.5.8
 -  `Angular Material Version:` 1.0.9

**Additional Information**:
 -  `Browser Type:  Chrome (Webkit)
 -  `Browser Version: *` 51
 -  `OS: *` Windows 10 x64
 -  `Stack Traces:` 
```
"TypeError: undefined is not a function
    at Array.map (native)
    at l.selectedLabels (https://ajax.googleapis.com/ajax/libs/angular_material/1.0.9/angular-material.min.js:11:25250)
    at Scope.$digest (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.js:17515:34)
    at https://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.js:17721:26
    at completeOutstandingRequest (https://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.js:5964:10)
    at https://ajax.googleapis.com/ajax/libs/angularjs/1.5.8/angular.js:6243:7"
```

**Comment by topherfangio** [comment]
This issue has already been fixed and will be included in the 1.1.0 release.

Here is your Codepen updated to use master (along with some minor layout tweaks): http://codepen.io/topherfangio/pen/XKxrmx

If you still experience issues after upgrading, please comment and we will reopen.

**Comment by CameronHall** [comment]
@topherfangio  Perfect.  

Thanks Champ.
